pkgs/by-name/ut/ut1999/package.nix

```diff
@@ -0,0 +1,122 @@
+{ lib, stdenv, requireFile, autoPatchelfHook, fetchurl,
+libgcc, wxGTK32, innoextract, makeWrapper,
+libGL, SDL2, openal, libmpg123, libxmp }:
+
+let
+  oldUnrealPatch = stdenv.mkDerivation {
+    name = "ut1999-oldunreal-patch";
+    version = "469d";
+    sourceRoot = ".";
+    src = fetchurl {
+      url = "https://github.com/OldUnreal/UnrealTournamentPatches/releases/download/v469d/OldUnreal-UTPatch469d-Linux-amd64.tar.bz2";
+      hash = "sha256-aoGzWuakwN/OL4+xUq8WEpd2c1rrNN/DkffI2vDVGjs=";
+    };
+
+    buildInputs = [
+      libgcc
+      stdenv.cc.cc
+      wxGTK32
+    ];
+
+    nativeBuildInputs = [
+      autoPatchelfHook
+      makeWrapper
+    ];
+
+    installPhase = ''
+      runHook preInstall
+
+      mkdir -p $out
+      ls -alh
+      cp -r ./* $out
+
+      runHook postInstall
+    '';
+
+    dontBuild = true;
+  };
+
+  unpackGog = stdenv.mkDerivation {
+    name = "ut1999-gog";
+    version = "2.0.0.5";
+
+    src = requireFile rec {
+      name = "setup_ut_goty_2.0.0.5.exe";  # FIXME: use official name
+      sha256 = "00v8jbqhgb1fry7jvr0i3mb5jscc19niigzjc989qrcp9pamghjc";
+      message = ''
+        Unreal Tournament 1999 requires the official GOG package, version 2.0.0.5.
+
+        Once you download the file, run the following command:
+
+        nix-prefetch-url file://\$PWD/${name}
+      '';
+    };
+
+    buildInputs = [
+      innoextract
+    ];
+
+    unpackPhase = ''
+      innoextract --extract --exclude-temp "$src"
+    '';
+
+    installPhase = ''
+      mkdir $out
+      cp -r app/* $out
+    '';
+
+    dontBuild = true;
+    dontFixup = true;
+  };
+in stdenv.mkDerivation {
+  name = "ut1999";
+  version = "2.0.0.5";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  installPhase = ''
+    runHook preInstall
+    mkdir $out
+
+    cp -r ${oldUnrealPatch}/* $out
+    chmod -R 755 $out
+
+    cp -r ${unpackGog}/Music $out
+    cp -r ${unpackGog}/Sounds $out
+    cp -n ${unpackGog}/Textures/* $out/Textures || true
+    cp -r ${unpackGog}/Maps $out
+
+    # Remove bundled libraries to use native versions instead
+    rm $out/System64/libfmod.so* \
+      $out/System64/libmpg123.so* \
+      $out/System64/libopenal.so* \
+      $out/System64/libSDL2* \
+      $out/System64/libxmp.so*
+
+    makeWrapper "$out/System64/ut-bin" "$out/bin/ut-bin" \
+      --suffix LD_LIBRARY_PATH : "${lib.strings.makeLibraryPath [
+        libGL
+        SDL2
+        openal
+        libmpg123
+        libxmp
+      ]}"
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Unreal Tournament GOTY (1999) with the OldUnreal patch.";
+    license = licenses.unfree;
+    platforms = [ "x86_64-linux" ];
+    maintainers = with maintainers; [ eliandoran ];
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    mainProgram = "ut-bin";
+  };
+
+  dontUnpack = true;
+  dontBuild = true;
+  dontFixup = true;
```

Comment:
Probably better as a `runCommand` again.

Reply:
@Atemu , my idea was to split the package into three sub-derivations:

1. The old-unreal patch which is downloaded from GitHub and needs ELF patching.
2. The gog data which uses `requireFile` to obtain the data and uses `innoextract` to unpack it.
3. The final derivation which puts the patch and the data together.

The idea was also to detach the `gog` data from the rest of the derivation, since it's theoretically possible to use the original GOTY ISO instead of the GOG data (I might consider adding it provided there is interest for it).

The only problem with this approach is that in the final derivation I copy over the data so I think I create some duplication in the nix store. The upside to this is that builds are faster, since we are not extracting the gog setup for each build (takes ~30s on my machine).

cc. @SuperSandro2000

Reply:
That sounds like a decent plan to me.

runCommand vs. mkDerivation is just a case of using the right tool for the job. runCommand is like a minimal mkDerivation where the intent is to only run one set of commands; without all the phases or even a C compiler. If you're overriding `installPhase` or `buildPhase` and disabling all other phases, you should be using runCommand instead.

If it's just data that does not need patching, symlink it instead.